numba/parfor.py

```diff
@@ -587,18 +587,41 @@ def replace_func():
                             self.typemap[g_np_var.name] = types.misc.Module(numpy)
                             g_np = ir.Global('np', numpy, loc)
                             g_np_assign = ir.Assign(g_np, g_np_var, loc)
+
                             typ_var = ir.Var(scope, mk_unique_var("$np_typ_var"), loc)
-                            self.typemap[typ_var.name] = types.DType(dtype)
+                            self.typemap[typ_var.name] = types.functions.NumberClass(dtype)
                             dtype_str = str(dtype)
                             if dtype_str == 'bool':
                                 dtype_str = 'bool_'
                             np_typ_getattr = ir.Expr.getattr(g_np_var, dtype_str, loc)
                             typ_var_assign = ir.Assign(np_typ_getattr, typ_var, loc)
-                            instr.value = typ_var
+
+                            dtype_attr_var = ir.Var(scope, mk_unique_var("$dtype_attr_var"), loc)
```

Comment:
I think a comment or two about the intent of this section of code would be useful, to include why it is being done and the NumPy equivalent expression it is representing?

Reply:
Done.